config/src/config/state_sync_config.rs

```diff
@@ -221,12 +229,302 @@ impl Default for AptosDataClientConfig {
 }
 
 impl ConfigSanitizer for StateSyncConfig {
-    /// Validate and process the state sync config according to the given node role and chain ID
     fn sanitize(
         _node_config: &mut NodeConfig,
-        _node_role: RoleType,
+        _node_type: NodeType,
         _chain_id: ChainId,
     ) -> Result<(), Error> {
-        Ok(()) // TODO: add validation of higher-level properties once we have variable configs
+        Ok(())
+    }
+}
+
+impl ConfigOptimizer for StateSyncConfig {
+    fn optimize(
+        node_config: &mut NodeConfig,
+        local_config_yaml: &Value,
+        node_type: NodeType,
+        chain_id: ChainId,
+    ) -> Result<bool, Error> {
+        // Optimize the driver and data streaming service configs
+        let modified_driver_config =
+            StateSyncDriverConfig::optimize(node_config, local_config_yaml, node_type, chain_id)?;
+        let modified_data_streaming_config = DataStreamingServiceConfig::optimize(
+            node_config,
+            local_config_yaml,
+            node_type,
+            chain_id,
+        )?;
+
+        Ok(modified_driver_config || modified_data_streaming_config)
+    }
+}
+
+impl ConfigOptimizer for StateSyncDriverConfig {
+    fn optimize(
+        node_config: &mut NodeConfig,
+        local_config_yaml: &Value,
+        _node_type: NodeType,
+        chain_id: ChainId,
+    ) -> Result<bool, Error> {
+        let state_sync_driver_config = &mut node_config.state_sync.state_sync_driver;
+        let local_driver_config_yaml = &local_config_yaml["state_sync"]["state_sync_driver"];
+
+        // Default to fast sync for all testnet nodes
```

Comment:
Aah, I can update the comment. It's because testnet is long enough now that pruning has kicked in so none of the nodes can access the data anymore (unless you manually add seeds)) 😆